### ISSUE TITLE
Upgrading to version 25 stop respecting config and prevents start

### ISSUE THREAD

**Issue by BoBeR182** [keycloak/keycloak#30777]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

After upgrading to version 25 on Arch, Keycloak fails to start.

### Version

25

### Regression

- [ ] The issue is a regression

### Expected behavior

Keycloak starts

### Actual behavior

```
Jun 25 20:41:36 host kc.sh[386275]: 2024-06-25 20:41:32,364 WARN  [org.key.qua.run.cli.Picocli] (main) The following run time non-cli options were found, but will be ignored during build time: kc.db-username, kc.db-password, kc.hostname, kc.https-certificate-file, kc.https-certificate-key-file, kc.proxy
Jun 25 20:41:51 host kc.sh[386628]: 2024-06-25 20:41:51,370 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to start server in (production) mode
Jun 25 20:41:51 host kc.sh[386628]: 2024-06-25 20:41:51,370 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Unable to start HTTP server
Jun 25 20:41:51 host kc.sh[386628]: 2024-06-25 20:41:51,370 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: java.lang.IllegalStateException: Unable to start the management interface
Jun 25 20:41:51 host kc.sh[386628]: 2024-06-25 20:41:51,370 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Unable to start the management interface
Jun 25 20:41:51 host kc.sh[386628]: 2024-06-25 20:41:51,370 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Address already in use
```

### How to Reproduce?

/etc/keycloak/keycloak.conf
```
# The database vendor.
db=mariadb
# The username of the database user.
db-username=keycloak
# The password of the database user.
db-password=REDACTED
# The full database JDBC URL. If not provided, a default URL is set based on the selected database vendor.
#db-url=jdbc:postgresql://localhost/keycloak

# HTTP
# The file path to a server certificate or certificate chain in PEM format.
https-certificate-file=/etc/letsencrypt/live/host.tld/fullchain.pem
# The file path to a private key in PEM format.
https-certificate-key-file=/etc/letsencrypt/live/host.tld/privkey.pem

# The proxy address forwarding mode if the server is behind a reverse proxy.
proxy=reencrypt

# Hostname for the Keycloak server.
hostname=keycloak.host.tld
```
Upgrade from 24 to 25

**Comment by vramik** [comment]
@BoBeR182 Thank you for the report, from the log you've provided it seems the port you're trying to use is not available. It often happens when there is already another instance of the server running. Please check whether it is the case. 

I'm closing the issue for now. Feel free to re-open with steps to reproduce it.

**Comment by BoBeR182** [comment]
I have it set to not serve on HTTP port 8080 and only on TLS port 9443, yet it ignores the config and attempt to bind to 8080 (which yes is in use)
This issue does not occure on 24 just 25 from the Arch repos.

**Comment by BoBeR182** [comment]
@vramik can you reopen this?

**Comment by crystalgreen** [comment]
I use v25.0.1 on Windows 11 and get a similar error:

```
ERROR: Failed to start server in (production) mode
ERROR: Unable to start HTTP server
ERROR: java.lang.IllegalStateException: Unable to start the management interface
ERROR: Unable to start the management interface
ERROR: Address already in use: bind
```

My relevant (default) config:
```
http-enabled=false
https-port=8443
```
The verbose log seems to indicate that port 8443 should be used but that port is definitely free:
`DEBUG [io.quarkus.micrometer.runtime.binder.vertx.VertxMeterBinderAdapter] (vert.x-acceptor-thread-0) Create HttpServerMetrics with options io.vertx.core.http.HttpServerOptions@475e0532 and address 0.0.0.0:8443`.

It all worked well on v24.

**Comment by vramik** [comment]
Duplicate of https://github.com/keycloak/keycloak/issues/31134, therefore keeping this one closed in favor of #31134
